test-data/unit/check-based-infer-function-types.test

```diff
@@ -267,3 +269,70 @@ class S:
     def f(self, f: int) -> str: ...
     @overload
     def f(self, f: str) -> int: ...
+
+
+[case testCheckUntypedDefs]
+# flags: --no-check-untyped-defs --allow-untyped-defs
+
+def foo(a=1):
+    1 + ""
+
+foo()  # E: Call to untyped function "foo" in typed context  [no-untyped-call]
+reveal_type(foo)  # E: Expression type contains "Any" (has type "(int) -> Untyped")  [no-any-expr] \
+                  # N: Revealed type is "def (a: int =) -> Untyped"
+
+def bar(a=1) -> None:
+    1 + ""  # E: Unsupported operand types for + ("int" and "str")  [operator]
+
+bar()
+reveal_type(bar)  # N: Revealed type is "def (a: int =)"
+
+
+[case inferPropertyTypes]
+# flags: --no-infer-function-types --no-default-return --no-check-untyped-defs
+class A:
+    @property
+    def f(self) -> int: ...
+    @f.setter
+    def f(self, value): ...
+    @f.deleter
+    def f(self): ...
+
+a = A()
+reveal_type(a.f)  # N: Revealed type is "int"
+a.f = ""  # E: Incompatible types in assignment (expression has type "str", variable has type "int")  [assignment]
+del a.f
+
+class B:
+    @property
+    def f(self): ...  # E: Function is missing a return type annotation  [no-untyped-def] \
+                      # N: Use "-> None" if function does not return a value \
+                      # E: Type of decorated function contains type "Any" ("(B) -> Untyped")  [no-any-decorated]
+    @f.setter
+    def f(self, value):
+        1 + ""
+    @f.deleter
+    def f(self): ...
+
+b = B()
+reveal_type(b.f)  # E: Expression has type "Untyped"  [no-any-expr] \
+                  # N: Revealed type is "Untyped"
+b.f = 1
+del b.f
+[builtins fixtures/property.pyi]
+
+
+[case testPropertyNoAnnotation]
+class A:
+    @property
+    def p(self): ...  # E: Property is missing a type annotation  [no-untyped-def]
+    @p.setter
+    def p(self, value): ...
+    @p.deleter
+    def p(self): ...
```

Comment:
can u add tests involving inferring types from overridden properties in supertypes

Reply:
Overloaded overrides aren't supported, but the non overloaded version should be fine.

Reply:
added